### PR TITLE
buildDotnetPackage/fetchNuGet: use `pname` instead of `name` and `baseName`

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
follow up of #144949

for `buildDotnetPackage` the variable `baseName` was renamed to `pname`.
i found, the naming of `fetchNuGet` use `name` and `version`. therefore i decided to rename it to `pname` & `version`.

sadly this PR turned out to be larger, then expected

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/dotnet-modules/python-language-server/deps.nix [hunk 1]
```diff
@@ -1,1231 +1 @@
-# This file is autogenerated.
-# To regenerate, run "create_deps.sh $PATH_TO_LANGUAGE_SERVER_CHECKOUT"
```
Comment (SuperSandro2000) [inline pkgs/development/dotnet-modules/python-language-server/deps.nix:1]: This file should not be empty.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
At some point we should cleanup all those extra fetchNugetDeps functions.

File: pkgs/development/dotnet-modules/python-language-server/deps.nix (hunk 1 above)
Comment (Stunkymonkey) [reply pkgs/development/dotnet-modules/python-language-server/deps.nix:1]: not sure what happened here. should be fixed.

File: pkgs/applications/blockchains/wasabibackend/deps.nix [hunk 2]
```diff
@@ -1,952 +1,955 @@
...
   nugetUrlBase = "https://www.nuget.org/api/v2/package";
-  fetchNuGet = { name, version, sha256 }: fetchurl { inherit sha256; url = "${nugetUrlBase}/${name}/${version}"; };
-in
-[
+  fetchNuGet = { pname, version, sha256 }:
+    fetchurl {
+      inherit sha256;
+      url = "${nugetUrlBase}/${pname}/${version}";
+    };
+in [
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/wasabibackend/deps.nix:9]: Can we use the top level function here, too?

Comment (Stunkymonkey) [reply pkgs/applications/blockchains/wasabibackend/deps.nix]: fine this way?

**Comment by Stunkymonkey** [comment]
can this be merged to `master`, or should i rebase it to `staging`?

**Comment by prusnak** [comment]
> can this be merged to `master`,

Let's aim for master. There are not a lot of rebuilds (11-100). Please rebase and resolve the conflicts.

@IvarWithoutBones can you please have a look? what do you think about this change?

**Comment by Stunkymonkey** [comment]
I removed the two commits about `wasabibackend`&`github-runner` and will adress the improvments in a separate PR, because this PR has a long diff.

File: pkgs/build-support/build-dotnet-module/default.nix [hunk 3]
```diff
@@ -68,9 +68,9 @@ assert nugetDeps == null -> throw "Defining the `nugetDeps` attribute is require
 
 let
   _nugetDeps = linkFarmFromDrvs "${name}-nuget-deps" (import nugetDeps {
-    fetchNuGet = { name, version, sha256 }: fetchurl {
-      name = "nuget-${name}-${version}.nupkg";
-      url = "https://www.nuget.org/api/v2/package/${name}/${version}";
+    fetchNuGet = { pname, version, sha256 }: fetchurl {
+      name = "nuget-${pname}-${version}.nupkg";
```
Comment (IvarWithoutBones) [inline pkgs/build-support/build-dotnet-module/default.nix:72]: Would you mind changing this? The `linkFarmFromDrvs`'s name already expresses that it contains nuget deps.
```suggestion
      name = "${pname}-${version}.nupkg";
```

This comment applies to `osu-lazer` and `jellyfin` as well.

**Review by IvarWithoutBones — COMMENTED** [review]
LGTM, i think this makes `buildDotnetPackage` derivations look a lot cleaner, I just have 1 (kind of unrelated) suggestion :)

Thanks for the ping btw! I appreciate it.

File: pkgs/build-support/build-dotnet-module/default.nix (hunk 3 above)
Comment (Stunkymonkey) [reply pkgs/build-support/build-dotnet-module/default.nix]: yes for sure. should be fixed now.
If you approve my changes please approve the PR.

**Comment by prusnak** [comment]
@GrahamcOfBorg build btcpayserver nbxplorer

**Review by prusnak — APPROVED** [review]

**Comment by IvarWithoutBones** [comment]
Thanks!

**Pull request merged**